app/controllers/reports_controller.rb

```diff
@@ -17,11 +17,34 @@ def update
     time_regs = TimeReg.for_report(client_ids_for_report, project_ids_for_report, user_ids_for_report, task_ids_for_report)
                        .where(date_worked: (@selected_start_date..@selected_end_date))
 
-    @structured_report_data = TimeRegsPresenter.new(time_regs).report_data(
-      title: nil,
-      keys: [ :client, :project, :task, :user ]
-    )
+    if @form_data.detailed_report
+
+      @structured_report_data = time_regs.group_by { |reg| reg[:date_worked] }.sort_by { |key| key  }
+
+      clients = Client.joins(:time_regs).where(time_regs: { id: time_regs }).distinct
+      projects = Project.joins(:time_regs).where(time_regs: { id: time_regs }).distinct
+      tasks = Task.joins(:time_regs).where(time_regs: { id: time_regs }).distinct
+      users = User.joins(:time_regs).where(time_regs: { id: time_regs }).distinct
+      total_billable_minutes = time_regs.joins(:project).where(project: { billable_project: true }).sum(:minutes)
+      total_minutes = time_regs.sum(:minutes)
+
+      @detailed_report_data = OpenStruct.new(
```

Comment:
I noticed that OpenStruct is being used here. In the context of this update method being called frequently with filter options, there might be minimal performance gains from using a simpler hash structure.

It's important to note that this doesn't have a noticeable impact in this specific case and would go unnoticed. I'm not requesting changes on this I'm just mentioning it for future considerations
Ref -> https://www.honeybadger.io/blog/how-openstruct-and-hashes-can-kill-performance/